packages/block-editor/src/store/actions.js

```diff
@@ -998,14 +998,17 @@ export const __unstableSplitSelection =
 
 		let tail = {
 			...blockB,
-			// Only preserve the original client ID if the end is different.
 			clientId:
 				blockA.clientId === blockB.clientId
 					? createBlock( blockB.name ).clientId
 					: blockB.clientId,
 			attributes: {
 				...blockB.attributes,
 				[ attributeKeyB ]: toHTMLString( { value: valueB } ),
+				// Add widget ID clearance for same-block splits
+				...( blockA.clientId === blockB.clientId && {
+					__internalWidgetId: undefined,
+				} ),
```

Comment:
I don't think adding a special case to the `block-editor` package is a good option. Why?

* The `block-editor` doesn't need to know about Widget editor specifics. The `__internalWidgetId` is an internal ID for that screen.
* What happens when we have more unique attributes like `__internalWidgetId`? Do we keep maintaining the list?